trino-lb/src/http_server/mod.rs

```diff
@@ -121,9 +123,17 @@ pub async fn start_http_server(
             "/v1/statement/executing/{query_id}/{slug}/{token}",
             delete(v1::statement::delete_trino_executing_statement),
         )
+        .route(
+            "/v1/trino-event-listener",
+            post(v1::trino_event_listener::post_trino_event_listener),
+        )
         .route("/ui/index.html", get(ui::index::get_ui_index))
         .route("/ui/query.html", get(ui::query::get_ui_query))
         .layer(TraceLayer::new_for_http())
+        // The Trino HTTP events (received at `/v1/trino-event-listener`) are compressed by default, so we need to need
+        // to be able to accept compressed content
+        .layer(RequestDecompressionLayer::new())
+        // Trino clients can ask for compressed data, so we should support compressing the response
```

Comment:
```suggestion
        // Allow requests from trino-lb to negotiate compression
        // The Trino HTTP events (received at `/v1/trino-event-listener`) are compressed by default, so we need to need
        // to be able to accept compressed content
        .layer(RequestDecompressionLayer::new())
        // Allow requests to trino-lb to negotiate compression
        // Trino clients can ask for compressed data, so we should support compressing the response
```

Reply:
I *think* `Allow requests from trino-lb to negotiate compression` is wrong, as we are talking about requests that Trino sends to trino-lb if I got it right

Reply:
Ah ok, but then again, how is that different to CompressionLayer?
In both cases, trino-lb is the server, and Trino Client or Trino is the client.

_Hopefully my understanding above helps you see my error in understanding_

Reply:
As per phone call, I understand what you mean now.
I have updated the suggestion to capture that.

Interestingly, there are two decompression layers:
- [DecompressionLayer](https://docs.rs/tower-http/latest/tower_http/decompression/struct.DecompressionLayer.html)
- [RequestDecompressionLayer](https://docs.rs/tower-http/latest/tower_http/decompression/struct.RequestDecompressionLayer.html)

While there is only one compression layer, [CompressionLayer](https://docs.rs/tower-http/latest/tower_http/compression/struct.CompressionLayer.html)